src/server/main.ts

```diff
@@ -25,17 +26,29 @@ app.post('/register', async (req, res) => {
   if (!userExists) {
     const writeUser = await databaseService.writeUser({ email, username, password });
     console.log(req.body);
-    res.json({ writeUser, status: 'Registration successful!' });
+    res.status(200).json({ writeUser, status: 'Registration successful!' });
   }
-  res.json('Username or email already exists');
+  res.status(200).json('Username or email already exists');
 });
 
-// TODO
-// post route for login
-// it checks that credentials are valid
-// if they aren't valid, return an error
-// if they are valid, generate a JWT and send it back to the user
-// log them in
+app.post('/login', async (req, res) => {
+  const { username, password } = req.body;
+
+  const isValidUser = await databaseService.verifyUser(username, password);
+
+  if (isValidUser) {
+    const getUserForJwt = (username: string): Promise<User | null> => {
```

Comment:
Why do you have this inline function defined?

Reply:
I think this is because he has no concept of a UserService type class, he only has the DatabaseService atm, and since he was going off my repo this function lived on the UserService